File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/create_view_details.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class CreateViewDetails(object):
    """
    The data to create a view.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new CreateViewDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param display_name:
            The value to assign to the display_name property of this CreateViewDetails.
        :type display_name: str

        :param catalog_key:
            The value to assign to the catalog_key property of this CreateViewDetails.
        :type catalog_key: str

        :param schema_key:
            The value to assign to the schema_key property of this CreateViewDetails.
        :type schema_key: str

        :param description:
            The value to assign to the description property of this CreateViewDetails.
        :type description: str

        :param view_text:
            The value to assign to the view_text property of this CreateViewDetails.
        :type view_text: str

        :param view_properties:
            The value to assign to the view_properties property of this CreateViewDetails.
        :type view_properties: list[oci.aidataplatform_dp.models.ViewProperty]

        :param view_fields:
            The value to assign to the view_fields property of this CreateViewDetails.
        :type view_fields: list[oci.aidataplatform_dp.models.ViewFieldDetails]

        """
        self.swagger_types = {
            'display_name': 'str',
            'catalog_key': 'str',
            'schema_key': 'str',
            'description': 'str',
            'view_text': 'str',
            'view_properties': 'list[ViewProperty]',
            'view_fields': 'list[ViewFieldDetails]'
        }

        self.attribute_map = {
            'display_name': 'displayName',
            'catalog_key': 'catalogKey',
            'schema_key': 'schemaKey',
            'description': 'description',
            'view_text': 'viewText',
            'view_properties': 'viewProperties',
            'view_fields': 'viewFields'
        }

        self._display_name = None
        self._catalog_key = None
        self._schema_key = None
        self._description = None
        self._view_text = None
        self._view_properties = None
        self._view_fields = None

    @property
    def display_name(self):
        """
        **[Required]** Gets the display_name of this CreateViewDetails.
        A user-friendly name. Has to be unique within the scope of the schema and is changeable.


        :return: The display_name of this CreateViewDetails.
        :rtype: str
        """
        return self._display_name

    @display_name.setter
    def display_name(self, display_name):
        """
        Sets the display_name of this CreateViewDetails.
        A user-friendly name. Has to be unique within the scope of the schema and is changeable.


        :param display_name: The display_name of this CreateViewDetails.
        :type: str
        """
        self._display_name = display_name

    @property
    def catalog_key(self):
        """
        **[Required]** Gets the catalog_key of this CreateViewDetails.
        The name of the catalog to which this view belongs.


        :return: The catalog_key of this CreateViewDetails.
        :rtype: str
        """
        return self._catalog_key

    @catalog_key.setter
    def catalog_key(self, catalog_key):
        """
        Sets the catalog_key of this CreateViewDetails.
        The name of the catalog to which this view belongs.


        :param catalog_key: The catalog_key of this CreateViewDetails.
        :type: str
        """
        self._catalog_key = catalog_key

    @property
    def schema_key(self):
        """
        **[Required]** Gets the schema_key of this CreateViewDetails.
        The name of the Schema to which this view belongs.


        :return: The schema_key of this CreateViewDetails.
        :rtype: str
        """
        return self._schema_key

    @schema_key.setter
    def schema_key(self, schema_key):
        """
        Sets the schema_key of this CreateViewDetails.
        The name of the Schema to which this view belongs.


        :param schema_key: The schema_key of this CreateViewDetails.
        :type: str
        """
        self._schema_key = schema_key

    @property
    def description(self):
        """
        Gets the description of this CreateViewDetails.
        The description of the view.


        :return: The description of this CreateViewDetails.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this CreateViewDetails.
        The description of the view.


        :param description: The description of this CreateViewDetails.
        :type: str
        """
        self._description = description

    @property
    def view_text(self):
        """
        **[Required]** Gets the view_text of this CreateViewDetails.
        The Query used to create the view.


        :return: The view_text of this CreateViewDetails.
        :rtype: str
        """
        return self._view_text

    @view_text.setter
    def view_text(self, view_text):
        """
        Sets the view_text of this CreateViewDetails.
        The Query used to create the view.


        :param view_text: The view_text of this CreateViewDetails.
        :type: str
        """
        self._view_text = view_text

    @property
    def view_properties(self):
        """
        Gets the view_properties of this CreateViewDetails.
        View Properties.


        :return: The view_properties of this CreateViewDetails.
        :rtype: list[oci.aidataplatform_dp.models.ViewProperty]
        """
        return self._view_properties

    @view_properties.setter
    def view_properties(self, view_properties):
        """
        Sets the view_properties of this CreateViewDetails.
        View Properties.


        :param view_properties: The view_properties of this CreateViewDetails.
        :type: list[oci.aidataplatform_dp.models.ViewProperty]
        """
        self._view_properties = view_properties

    @property
    def view_fields(self):
        """
        Gets the view_fields of this CreateViewDetails.
        Columns for view.


        :return: The view_fields of this CreateViewDetails.
        :rtype: list[oci.aidataplatform_dp.models.ViewFieldDetails]
        """
        return self._view_fields

    @view_fields.setter
    def view_fields(self, view_fields):
        """
        Sets the view_fields of this CreateViewDetails.
        Columns for view.


        :param view_fields: The view_fields of this CreateViewDetails.
        :type: list[oci.aidataplatform_dp.models.ViewFieldDetails]
        """
        self._view_fields = view_fields

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/CreateViewDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * The data to create a view.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=CreateViewDetails.Builder.class)

public final class CreateViewDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"displayName", "catalogKey", "schemaKey", "description", "viewText", "viewProperties", "viewFields"})
    public CreateViewDetails(String displayName, String catalogKey, String schemaKey, String description, String viewText, java.util.List<ViewProperty> viewProperties, java.util.List<ViewFieldDetails> viewFields) {
        super();
        this.displayName = displayName;
        this.catalogKey = catalogKey;
        this.schemaKey = schemaKey;
        this.description = description;
        this.viewText = viewText;
        this.viewProperties = viewProperties;
        this.viewFields = viewFields;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * A user-friendly name. Has to be unique within the scope of the schema and is changeable.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("displayName")
        private String displayName;

                /**
         * A user-friendly name. Has to be unique within the scope of the schema and is changeable.
         * @param displayName the value to set
         * @return this builder
         **/
        

        public Builder displayName(String displayName) {
        this.displayName = displayName;
        return this;
        }
            /**
     * The name of the catalog to which this view belongs.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("catalogKey")
        private String catalogKey;

                /**
         * The name of the catalog to which this view belongs.
         * @param catalogKey the value to set
         * @return this builder
         **/
        

        public Builder catalogKey(String catalogKey) {
        this.catalogKey = catalogKey;
        return this;
        }
            /**
     * The name of the Schema to which this view belongs.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("schemaKey")
        private String schemaKey;

                /**
         * The name of the Schema to which this view belongs.
         * @param schemaKey the value to set
         * @return this builder
         **/
        

        public Builder schemaKey(String schemaKey) {
        this.schemaKey = schemaKey;
        return this;
        }
            /**
     * The description of the view.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("description")
        private String description;

                /**
         * The description of the view.
         * @param description the value to set
         * @return this builder
         **/
        

        public Builder description(String description) {
        this.description = description;
        return this;
        }
            /**
     * The Query used to create the view.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("viewText")
        private String viewText;

                /**
         * The Query used to create the view.
         * @param viewText the value to set
         * @return this builder
         **/
        

        public Builder viewText(String viewText) {
        this.viewText = viewText;
        return this;
        }
            /**
     * View Properties.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("viewProperties")
        private java.util.List<ViewProperty> viewProperties;

                /**
         * View Properties.
         * @param viewProperties the value to set
         * @return this builder
         **/
        

        public Builder viewProperties(java.util.List<ViewProperty> viewProperties) {
        this.viewProperties = viewProperties;
        return this;
        }
            /**
     * Columns for view.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("viewFields")
        private java.util.List<ViewFieldDetails> viewFields;

                /**
         * Columns for view.
         * @param viewFields the value to set
         * @return this builder
         **/
        

        public Builder viewFields(java.util.List<ViewFieldDetails> viewFields) {
        this.viewFields = viewFields;
        return this;
        }


        public CreateViewDetails build() {
            CreateViewDetails model = new CreateViewDetails(this.displayName
                    , this.catalogKey
                    , this.schemaKey
                    , this.description
                    , this.viewText
                    , this.viewProperties
                    , this.viewFields);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(CreateViewDetails model) {
                this.displayName(model.getDisplayName());
    this.catalogKey(model.getCatalogKey());
    this.schemaKey(model.getSchemaKey());
    this.description(model.getDescription());
    this.viewText(model.getViewText());
    this.viewProperties(model.getViewProperties());
    this.viewFields(model.getViewFields());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * A user-friendly name. Has to be unique within the scope of the schema and is changeable.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("displayName")
    private final String displayName;

        /**
     * A user-friendly name. Has to be unique within the scope of the schema and is changeable.
     * @return the value
     **/
    
    public String getDisplayName() {
        return displayName;
    }


        /**
     * The name of the catalog to which this view belongs.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("catalogKey")
    private final String catalogKey;

        /**
     * The name of the catalog to which this view belongs.
     * @return the value
     **/
    
    public String getCatalogKey() {
        return catalogKey;
    }


        /**
     * The name of the Schema to which this view belongs.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("schemaKey")
    private final String schemaKey;

        /**
     * The name of the Schema to which this view belongs.
     * @return the value
     **/
    
    public String getSchemaKey() {
        return schemaKey;
    }


        /**
     * The description of the view.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

        /**
     * The description of the view.
     * @return the value
     **/
    
    public String getDescription() {
        return description;
    }


        /**
     * The Query used to create the view.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("viewText")
    private final String viewText;

        /**
     * The Query used to create the view.
     * @return the value
     **/
    
    public String getViewText() {
        return viewText;
    }


        /**
     * View Properties.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("viewProperties")
    private final java.util.List<ViewProperty> viewProperties;

        /**
     * View Properties.
     * @return the value
     **/
    
    public java.util.List<ViewProperty> getViewProperties() {
        return viewProperties;
    }


        /**
     * Columns for view.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("viewFields")
    private final java.util.List<ViewFieldDetails> viewFields;

        /**
     * Columns for view.
     * @return the value
     **/
    
    public java.util.List<ViewFieldDetails> getViewFields() {
        return viewFields;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("CreateViewDetails(");
        sb.append("displayName=").append(String.valueOf(this.displayName));
        sb.append(", catalogKey=").append(String.valueOf(this.catalogKey));
        sb.append(", schemaKey=").append(String.valueOf(this.schemaKey));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", viewText=").append(String.valueOf(this.viewText));
        sb.append(", viewProperties=").append(String.valueOf(this.viewProperties));
        sb.append(", viewFields=").append(String.valueOf(this.viewFields));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof CreateViewDetails)) {
            return false;
        }

        CreateViewDetails other = (CreateViewDetails) o;
        return java.util.Objects.equals(this.displayName, other.displayName) &&
            java.util.Objects.equals(this.catalogKey, other.catalogKey) &&
            java.util.Objects.equals(this.schemaKey, other.schemaKey) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.viewText, other.viewText) &&
            java.util.Objects.equals(this.viewProperties, other.viewProperties) &&
            java.util.Objects.equals(this.viewFields, other.viewFields);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.displayName == null ? 43 : this.displayName.hashCode());
        result = (result * PRIME) + (this.catalogKey == null ? 43 : this.catalogKey.hashCode());
        result = (result * PRIME) + (this.schemaKey == null ? 43 : this.schemaKey.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.viewText == null ? 43 : this.viewText.hashCode());
        result = (result * PRIME) + (this.viewProperties == null ? 43 : this.viewProperties.hashCode());
        result = (result * PRIME) + (this.viewFields == null ? 43 : this.viewFields.hashCode());
        return result;
    }


}
